### PR TITLE
Functionality: CRUD methods

### PR DESCRIPTION
**Pull request by TracyMuso**
- Implemented a function for adding a new task (add a new element to the array).
- Implemented a function for deleting a task (remove an element from the array).
- Implemented a function for editing task descriptions.
- All changes to the To Do List are be saved in local storage.
- Index of remaining items is updated after removing one task.

### CODE REVIEW

File: src/assets/create.js [hunk 1]
```diff
@@ -1,36 +1,28 @@
...
-      toDo.setAttribute('type', 'checkbox');
-      toDo.id = 'id1';
-      const dots = document.createElement('i');
-      dots.className = 'bi bi-three-dots-vertical';
-      listCont.append(toDo, label, dots);
+      listCont.setAttribute('data-index', task.index);
+      listCont.innerHTML = `
+      <label for"id1"></label>
+      <input class="input" type="checkbox" id="id1">
+         <span class="span" contenteditable="true">${task.description}</span>
```
Comment (ahmetbozaci) [inline src/assets/create.js]: - [ ] I can edit the to-do item but it is not changing in the local storage. Kindly fix it.
image

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# ♻️♻️♻️ STATUS: Changes Required  ♻️♻️♻️


 @elfin-git  ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.

___ 

### 💯  Highlights 💯
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
- I can add and remove to-do items.✔️
- If ı  remove todo item index updated.✔️
- All todo items are saved in the localstorage.✔️
___


### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
image

___

### Cheers and Happy coding! 💻 👏
almost

___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

File: dist/index.html [hunk 2]
```diff
@@ -14,10 +14,12 @@ <h3>Daily to-do</h3>
             <img src="/26fbf6351e01a1e739cb.png" class="refresh" alt="refresh icon">
         </div>
         <div id="new-item">
-            <input id="add-new" type="text" placeholder="Add new list item" >
-            <button type="button" id="plus"><i class="bi bi-plus"></i></button>
+            <label for="add-new"></label>
+            <input id="add-new" type="text" placeholder="Add new list item" required>
+            <button type="button" class="insert"><i class="bi bi-plus"></i></button>
         </div>
         <div id="list-items">
```
Comment (degisew) [inline dist/index.html]: Kindly add a proper indentation.

File: src/index.js [hunk 3]
```diff
@@ -1,8 +1,66 @@
 import './style.css';
+import List from './assets/obj.js';
 import UI from './assets/create.js';
+import Storage from './assets/store.js';
 import refreshIcon from './assets/Refresh_icon.png';
 
 const myIcon = new Image();
 myIcon.src = refreshIcon;
```
Comment (degisew) [inline src/index.js:8]: make sure you put a blank line for reason.

**Review by degisew — COMMENTED** [review]
Hi @elfin-git 
Great job so far 👏 👏 

You are almost there but there are still some issues that going to be fixed for your next project.

### Required changes ♻️ 
check the comment under the review.

**Review by gtekle — CHANGES_REQUESTED** [review]
Hi @elfin-git 👋 , 

# STATUS: Changes Requested 🔴 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
this close

 ## Highlights
✔️ Meaningful commit messages
✔️ Good use of Github flow
✔️ Descriptive PR summary
✔️ Professional ReadMe

 ## Required Changes ♻️
_Check the comments under the review._

- [ ] As stated by the previous reviewer, kindly save the updated task description into localStorage. Right now, the update is lost if you refresh the page. You may also check localStorage under applications tab in Chrome dev tools. 
image


## Optional suggestions

- [ OPTIONAL ] In future projects, kindly use more descriptive branch names. In this case, for example, `functionality` is not a descriptive enough branch name. You could have used some thing like `feature/task-crud-operations`

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by TracyMuso** [comment]
@gtekle , It is working fine on my end, let me try opening it on chrome, I am currently using firefox

**Comment by TracyMuso** [comment]
The event listener for the task has been changed, so with every change you make, the local storage is updated, please reload your local storage if you are on chrome. For now, everything is working fine on both  firefox and chrome

**Review by elmar8287 — APPROVED** [review]
Hi, @elfin-git.

# APPROVED  👍 🥇 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

Congratulations! 🎉 

### Highlights

- Actually, local storage is working 👍 🥇 

### Optional suggestions

- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
### Please, remember to tag me (@elmar8287) in your question so I can receive the notification 👍 
giphy

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**